File: src/main/java/com/study/platform/global/constant/KafkaConstants.java
package com.study.platform.global.constant;

import lombok.experimental.UtilityClass;

@UtilityClass
public class KafkaConstants {

    public static final String NOTIFICATION_TOPIC = "notification";
    public static final String NOTIFICATION_GROUP = "notification-group";

    public static final String POST_SYNC_TOPIC = "post-sync";
    public static final String POST_SYNC_GROUP = "post-sync-group";

    public static final String POST_SYNC_DLT_TOPIC = "post-sync.DLT";
    public static final String POST_SYNC_DLT_GROUP = "post-sync-dlt-group";
}

File: common/src/main/kotlin/com/study/platform/global/constant/KafkaConstants.kt
package com.study.platform.global.constant

object KafkaConstants {
    const val NOTIFICATION_TOPIC = "notification"
    const val NOTIFICATION_GROUP = "notification-group"

    const val POST_SYNC_TOPIC = "post-sync"
    const val POST_SYNC_GROUP = "post-sync-group"

    const val POST_SYNC_DLT_TOPIC = "post-sync.DLT"
    const val POST_SYNC_DLT_GROUP = "post-sync-dlt-group"

    const val NOTIFICATION_DLT_TOPIC = "notification.DLT"
    const val NOTIFICATION_DLT_GROUP = "notification-dlt-group"
    const val MAX_DLT_RETRY = 3
}
